// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Contains parameters for a network simulation model.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package media.cast.proto;

message NetworkSimulationModel {
  optional NetworkSimulationModelType type = 1;
  optional IPPModel ipp = 2;
}

enum NetworkSimulationModelType {
  // Network simulation based on interrupted poisson process.
  INTERRUPTED_POISSON_PROCESS = 1;

  // No network simulation.
  NO_SIMULATION = 2;
}

message IPPModel {
  optional double coef_burstiness = 1;
  optional double coef_variance = 2;
  repeated double average_rate = 3;
}
